File: python/pycdas/test/floatClient.py
#
#   Hello World client in Python
#   Connects REQ socket to tcp://localhost:5555
#   Sends "Hello" to server, expects "World" back
#

import zmq, numpy as np

context = zmq.Context()

#  Socket to talk to server
print("Connecting to hello world server")
socket = context.socket(zmq.REQ)
socket.connect("tcp://localhost:5555")
# val byte_data = input_array.toUcarFloatArray.getDataAsByteBuffer().array()

databuff = np.getbuffer( np.array([ 2.5, 3.1, 1.1, 0.1]) )
data = [ databuff[i] for i in range(0,4) ]
print("Sending request %d %d %d %d" % ( data[0], data[1], data[2], data[3] ) )
socket.send( data )

#  Get the reply.
message = socket.recv()
print("Received reply [ %s ]" % (message))

File: src/main/java/nasa/nccs/cdas/workers/test/floatClient.java
package nasa.nccs.cdas.workers.test;

//
//  Hello World client in Java
//  Connects REQ socket to tcp://localhost:5555
//  Sends "Hello" to server, expects "World" back
//

import org.zeromq.ZMQ;

public class floatClient {

    public static void run( byte[] data ) {
        ZMQ.Context context = ZMQ.context(1);

        //  Socket to talk to server
        System.out.println("Connecting to hello world server…");

        ZMQ.Socket requester = context.socket(ZMQ.REQ);
        requester.connect("tcp://localhost:5555");

        System.out.println("Sending Data ");
        requester.send(data, 0);

        byte[] reply = requester.recv(0);
        System.out.println( "Received " + new String(reply) );

        requester.close();
        context.term();
    }
}